clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -705,16 +715,25 @@ public Node leastLoadedNode(long now) {
         Node foundCanConnect = null;
         Node foundReady = null;
 
+        boolean atLeastOneNodeConnected = false;
+
         int offset = this.randOffset.nextInt(nodes.size());
         for (int i = 0; i < nodes.size(); i++) {
             int idx = (offset + i) % nodes.size();
             Node node = nodes.get(idx);
+
+            if (!atLeastOneNodeConnected
+                    && connectionStates.isReady(node.idString(), now)
+                    && selector.isChannelReady(node.idString())) {
+                atLeastOneNodeConnected = true;
```

Comment:
Can we add one unit test for the various combinations to verify the returned values for LeastLoadedNode?

Reply:
Right, I added some unit testing